### PR TITLE
[Bugfix] fix accuracy problem for quantized deepseek models

### PR DESCRIPTION
**Pull request by linfeng-yuan**
### What this PR does / why we need it?

The root cause of the bug is that numerical computations involving NaNs cannot eliminate them. We addressed it by using `masked_fill_` to eliminate NaNs while avoiding memory-wasting `torch.where` approach.

### Does this PR introduce _any_ user-facing change?
Not invovled.

### How was this patch tested?
This patch was tested with vllm v0.8.5 and vllm-ascend master. I run deepseek_v3 model with offline inference scripts (examples/dp_offline/run_dp.sh & data_parallel.py).

### CODE REVIEW

File: vllm_ascend/quantization/w8a8_dynamic.py [hunk 1]
```diff
@@ -285,8 +285,9 @@ def fused_experts(hidden_states: torch.Tensor,
         valid_token_mask = torch.arange(
             0, sorted_token_indices.shape[0],
             device=device).unsqueeze(1) < num_valid_tokens
-        down_out_list.mul_(valid_token_mask)
-        final_hidden_states.index_add_(0, sorted_token_indices, down_out_list)
+        valid_output = torch.where(valid_token_mask, down_out_list,
```
Comment (ApsarasX) [inline vllm_ascend/quantization/w8a8_dynamic.py]: This way of using `torch.where` will significantly increase GPU memory usage. Can it be rewritten in another way? For example, using `masked_fill` like the following code?"

```
down_out_list  = down_out_list.masked_fill_(~valid_token_mask, 0).to(dtype)
```

Comment (linfeng-yuan) [reply vllm_ascend/quantization/w8a8_dynamic.py]: > This way of using `torch.where` will significantly increase GPU memory usage. Can it be rewritten in another way? For example, using `masked_fill` like the following code?"
> 
> ```
> down_out_list  = down_out_list.masked_fill_(~valid_token_mask, 0).to(dtype)
> ```

Thanks for suggestions! I think it must works in eager mode, and will verify this modifications in graph mode. Looks pretty good to me~

**Review by wangxiyuan — APPROVED** [review]

File: vllm_ascend/quantization/w8a8_dynamic.py (hunk 1 above)
Comment (wangxiyuan) [reply vllm_ascend/quantization/w8a8_dynamic.py]: it's good add some comment abut the change in the code file
